iceoryx2-bb/cxx/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache Software License 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+# which is available at https://opensource.org/licenses/MIT.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+cmake_minimum_required(VERSION 3.22)
+
+project(iceoryx2-bb VERSION ${IOX2_VERSION_STRING} LANGUAGES CXX)
+
+set(PREFIX iceoryx2/v${CMAKE_PROJECT_VERSION})
+
+# include only lib -> includes are installed only once despite being used in static lib as well as shared lib
+
+add_library(iceoryx2-bb-containers INTERFACE)
+add_library(iceoryx2-cxx::iceoryx2-bb-containers ALIAS iceoryx2-bb-containers)
+
+target_include_directories(iceoryx2-bb-containers
+    INTERFACE
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
```

Comment:
Out of curiosity. We used always `CMAKE_CURRENT_SOURCE_DIR` but it's good to get some feedback from outside of our bubble. Has `PROJECT_SOURCE_DIR` advantages over `CMAKE_CURRENT_SOURCE_DIR`? I'm happy to change it everywhere else to `PROJECT_SOURCE_DIR`

Reply:
`CMAKE_CURRENT_SOURCE_DIR` is always the very top of the source tree that CMake is aware of (so the directory passed to `cmake -S ...`), `PROJECT_SOURCE_DIR` is the directory where the last `CMakeLists.txt` with a `project` command is located. If you have nested projects like iceoryx, you almost alwas want `PROJECT_SOURCE_DIR`, otherwise you will end up with a different path when building the projects standalone. In general the legitimate uses for `CMAKE_CURRENT_SOURCE_DIR` are very few.